src/Compilers/CSharp/Test/Emit/Emit/EmitMetadataTests.cs

```diff
@@ -2936,5 +2936,576 @@ public class Child : Parent, IParent
+                // System.Console.Write("bb");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""bb""").WithArguments("System.Text.Encoding", "GetString").WithLocation(2, 22),
+                // (3,22): error CS0656: Missing compiler required member 'System.Text.Encoding.get_UTF8'
+                // System.Console.Write("ccc");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""ccc""").WithArguments("System.Text.Encoding", "get_UTF8").WithLocation(3, 22),
+                // (3,22): error CS0656: Missing compiler required member 'System.Text.Encoding.GetString'
+                // System.Console.Write("ccc");
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""ccc""").WithArguments("System.Text.Encoding", "GetString").WithLocation(3, 22));
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithFeature("experimental-data-section-string-literals", "3"));
+            comp.MakeMemberMissing(WellKnownMember.System_Text_Encoding__get_UTF8);
+            comp.MakeMemberMissing(WellKnownMember.System_Text_Encoding__GetString);
+            CompileAndVerify(comp, expectedOutput: "abbccc").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DataSectionStringLiterals_Threshold()
+        {
+            var source = """
+                System.Console.Write("a");
+                System.Console.Write("bb");
+                System.Console.Write("ccc");
+                """;
+
+            var expectedOutput = "abbccc";
+
+            var expectedIl = """
+                {
+                  // Code size       31 (0x1f)
+                  .maxstack  1
+                  IL_0000:  ldstr      "a"
+                  IL_0005:  call       "void System.Console.Write(string)"
+                  IL_000a:  ldstr      "bb"
+                  IL_000f:  call       "void System.Console.Write(string)"
+                  IL_0014:  ldstr      "ccc"
+                  IL_0019:  call       "void System.Console.Write(string)"
+                  IL_001e:  ret
+                }
+                """;
+
+            var verifier = CompileAndVerify(source, expectedOutput: expectedOutput)
+                .VerifyDiagnostics()
+                .VerifyIL("<top-level-statements-entry-point>", expectedIl);
+            Assert.Null(verifier.Compilation.DataSectionStringLiteralThreshold);
+
+            foreach (var feature in new[] { "off", null })
+            {
+                verifier = CompileAndVerify(source,
+                    parseOptions: TestOptions.Regular.WithFeature("experimental-data-section-string-literals", null),
+                    expectedOutput: expectedOutput)
+                    .VerifyDiagnostics()
+                    .VerifyIL("<top-level-statements-entry-point>", expectedIl);
+                Assert.Null(verifier.Compilation.DataSectionStringLiteralThreshold);
```

Comment:
>Assert.Null(verifier.Compilation.DataSectionStringLiteralThreshold); [](http://example.com/codeflow?start=16&length=68)

It feels somewhat surprising that `null` is not treated as an unrecognized input, defaulting to threshold == 100

Reply:
`null` is commonly treated as "not specified" by feature flags - see:

https://github.com/dotnet/roslyn/blob/f68a47ab0e9137b21db4fc8409e542a314cc0272/src/Compilers/Core/Portable/Compilation/Compilation.cs#L3506-L3510

I think it would be confusing if this feature behaved differently.

Reply:
Consider explicitly stating this behavior in the spec